daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/IdString.scala

```diff
@@ -337,17 +341,25 @@ private[data] final class IdStringImpl extends IdString {
   override val ContractIdString: StringModule[ContractIdString] =
     new MatchingStringModule("Daml-LF Contract ID", """#[\w._:\-#/ ]{0,254}""")
 
-  /** Identifiers for participant node users consist of ASCII digits and lower-case alphabetic characters,
-    *  hyphens, underscores, and dots, and satisfy the following  rules:
-    *  1. The characters `-._` never follow each other.
-    *  2. The characters `-._` neither occur at the start nor at the end of the user name.
-    *  Thus 'john.doe1' is a valid user-name, while 'john..doe1', 'john.-doe1', and '-john.doe' are not.
+  /** Identifiers for applications as used in custom tokens and requests.
+    * They used to be equal to [[LedgerString]], but were made to also allow '@' and '|'
+    * when introducing participant users to ensure that they are a superset of [[UserId]].
+    */
+  override type ApplicationId = String
+  override val ApplicationId: ConcatenableStringModule[LedgerString, HexString] =
+    new ConcatenableMatchingStringModule("Application ID", "._:-#/@| ", 255)
+
+  /** Identifiers for participant node users are non-empty strings that consist of ASCII alphanumeric characters and
```

Comment:
maybe a good idea to add a link to the auth0 restrictions so the restrictions feel a bit less arbitrary?

Reply:
Good point. Reading https://auth0.com/docs/authenticate/database-connections/require-username#allowed-characters, I found that Auth0 is also quite a bit more liberal in what it allows.

Reply:
I'll adjust accordingly.